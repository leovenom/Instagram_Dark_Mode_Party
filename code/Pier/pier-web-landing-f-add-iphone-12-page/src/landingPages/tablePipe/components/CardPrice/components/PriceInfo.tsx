import * as React from "react";

import Text from "ui/Text";
import data from "../data";

import {
  getArrowsUiOrientation,
  getMonthlyValue,
  getInsuredValue,
} from "../../utils";
import { CardRow, CardRowItem } from "../styles";

interface PriceInfo {
  smartphoneSelected: {
    plansValues: object;
    name: string;
    memory: string;
    model: {
      name: string;
    };
  };
  variant: string;
}

const PriceInfo: React.FC<PriceInfo> = ({ smartphoneSelected, variant }) => {
  const arrowsOrientation = getArrowsUiOrientation({
    smartphoneSelected,
    variant,
  });

  return (
    <>
      <CardRow>
        <CardRowItem>
          <Text variant="bodySmall" color="gray600">
            {data?.plan}
          </Text>
        </CardRowItem>
        <CardRowItem>
          <Text
            variant="bodySmallest"
            bold
            transform="uppercase"
            color="primary400"
          >
            {data?.economic}
          </Text>
        </CardRowItem>
        <CardRowItem>
          <Text
            variant="bodySmallest"
            bold
            transform="uppercase"
            color="primary400"
          >
            {data?.premium}
          </Text>
        </CardRowItem>
      </CardRow>

      <CardRow>
        <CardRowItem>
          <Text variant="bodySmall" color="gray600">
            {data?.monthlyPayment}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.monthly?.economic}>
          <Text variant="bodySmall" color="text">
            {getMonthlyValue("economic", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.monthly?.premium}>
          <Text variant="bodySmall" color="text">
            {getMonthlyValue("premium", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
      </CardRow>

      <CardRow>
        <CardRowItem>
          <Text variant="bodySmall" color="gray600">
            {data?.totalInsuranceValue}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.insuredValue?.economic}>
          <Text variant="bodySmall" color="text">
            {getInsuredValue("economic", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
        <CardRowItem arrow={arrowsOrientation?.insuredValue?.premium}>
          <Text variant="bodySmall" color="text">
            {getInsuredValue("premium", smartphoneSelected)?.formatted}
          </Text>
        </CardRowItem>
      </CardRow>

      <Text variant="body" align="center">
        {data.insuranceValueUsed}{" "}
        <strong>
          {getInsuredValue("premium", smartphoneSelected)?.formatted}
        </strong>
      </Text>
    </>
  );
};

export default PriceInfo;
