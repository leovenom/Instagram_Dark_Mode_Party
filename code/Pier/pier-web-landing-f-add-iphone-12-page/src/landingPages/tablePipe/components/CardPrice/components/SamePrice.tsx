import Text from "ui/Text";
import data from "../data";

import { ImgWoman } from "../styles";

const SamePrice = () => (
  <>
    <ImgWoman
      src={data.samePrice.imageSrc}
      alt={data.samePrice.imageDescription}
    />
    <Text
      variant="tag"
      bold
      transform="uppercase"
      align="center"
      color="primary400"
      mb={10}
    >
      {data.samePrice.title}
    </Text>
    <Text variant="bodySmall" align="center" color="gray600">
      {data.samePrice.description}
    </Text>
  </>
);

export default SamePrice;
