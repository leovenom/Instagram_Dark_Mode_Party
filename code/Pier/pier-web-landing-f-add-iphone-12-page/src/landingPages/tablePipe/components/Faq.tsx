import Grid from "ui/Grid";
import Collapse from "ui/Collapse";
import Section from "landingPages/sharedSections/Section";
import { smartphoneTracker } from "helpers/mixpanelTracker";

import {
  StyledTitleGridItem,
  StyledText,
  StyledCollapseGridItem,
  StyledLink,
} from "../styles";

import data from "../data";

const Faq = () => (
  <Section bg="white" id="section-faq" pt={0} pb={110}>
    <Grid>
      <StyledTitleGridItem m={2} t={8} d={4}>
        <StyledText mb={10} variant="sectionSecondary">
          {data.faq.title}
        </StyledText>
        <StyledText variant="body" color="gray">
          {data.faq.subtitle}
        </StyledText>
      </StyledTitleGridItem>
      <StyledCollapseGridItem m={2} t={8} d={8}>
        {data.faq.questions.map(({ question, answer }) => (
          <Collapse key={question}>
            <Collapse.Title>{question}</Collapse.Title>
            <Collapse.Content>{answer}</Collapse.Content>
          </Collapse>
        ))}
        <StyledLink
          withArrow
          href="https://ajuda.pier.digital/pt-BR/articles/2887266-atualizacao-da-tabela-pipe"
          target="_blank"
          rel="noopener"
          onClick={() =>
            smartphoneTracker.trackButton("Tabela Pipe See More Answers")
          }
        >
          {data.faq.link}
        </StyledLink>
      </StyledCollapseGridItem>
    </Grid>
  </Section>
);

export default Faq;
