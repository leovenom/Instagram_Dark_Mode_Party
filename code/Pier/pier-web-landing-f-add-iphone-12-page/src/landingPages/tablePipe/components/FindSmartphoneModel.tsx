import * as React from "react";

import Text from "ui/Text";
import Button from "ui/Button";

import { smartphoneTracker } from "helpers/mixpanelTracker";
import data from "../data";

interface FindSmartphoneModelProps {
  smartphoneSelected: {
    plansValues: object;
    name: string;
    memory: string;
    model: {
      name: string;
    };
  };
  toggleSmartphoneSelector: () => void;
}

const FindSmartphoneModel: React.FC<FindSmartphoneModelProps> = ({
  toggleSmartphoneSelector,
  smartphoneSelected,
}) => {
  if (smartphoneSelected) {
    return null;
  }
  const handleOpenModalSmartphoneModels = () => {
    smartphoneTracker.trackButton("Tabela Pipe Prices");
    toggleSmartphoneSelector();
  };

  return (
    <>
      <Text variant="section" align="center" mb={24}>
        {data.findSmartphoneModel.title}
      </Text>
      <Text variant="body" align="center" mb={40}>
        {data.findSmartphoneModel.subtitle}
      </Text>
      <Button size="small" onClick={handleOpenModalSmartphoneModels}>
        {data.findSmartphoneModel.cta}
      </Button>
    </>
  );
};

export default FindSmartphoneModel;
