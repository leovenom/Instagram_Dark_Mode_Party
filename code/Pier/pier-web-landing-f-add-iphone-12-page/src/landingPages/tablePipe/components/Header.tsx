import HeroCenteredContent from "landingPages/sharedSections/heroSections/HeroCenteredContent";
import data from "../data";

const Header = () => (
  <HeroCenteredContent
    bg="orange400"
    title={data.header.title}
    subtitle={data.header.subtitle}
    imageSrc={data.header.imageSrc}
    imageDescription={data.header.imageDescription}
    withoutTabs
  />
);

export default Header;
