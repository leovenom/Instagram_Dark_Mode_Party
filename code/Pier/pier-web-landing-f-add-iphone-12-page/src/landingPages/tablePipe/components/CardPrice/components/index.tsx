import PriceInfo from "./PriceInfo";
import SamePrice from "./SamePrice";

export { PriceInfo, SamePrice };
