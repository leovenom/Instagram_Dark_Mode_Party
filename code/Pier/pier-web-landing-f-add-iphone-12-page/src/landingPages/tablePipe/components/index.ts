import Header from "./Header";
import Instructions from "./Instructions";
import FindSmartphoneModel from "./FindSmartphoneModel";
import Faq from "./Faq";
import ShowPrices from "./ShowPrices";

export { Header, Instructions, FindSmartphoneModel, Faq, ShowPrices };
