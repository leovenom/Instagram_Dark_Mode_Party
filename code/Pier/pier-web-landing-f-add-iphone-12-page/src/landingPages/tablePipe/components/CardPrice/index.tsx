import CardPrice from "./CardPrice";

export default CardPrice;
