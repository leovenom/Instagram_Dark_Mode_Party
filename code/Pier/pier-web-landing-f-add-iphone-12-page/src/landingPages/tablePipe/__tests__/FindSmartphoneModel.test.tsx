import { fireEvent } from "@testing-library/react";

import { smartphoneTracker } from "helpers/mixpanelTracker";
import { renderWithTheme } from "helpers/testUtils";
import { FindSmartphoneModel } from "../components";

jest.mock("helpers/mixpanelTracker");
const toggleSmartphoneSelector = jest.fn();
const renderComponent = ({ smartphoneSelected }) =>
  renderWithTheme(
    <FindSmartphoneModel
      toggleSmartphoneSelector={toggleSmartphoneSelector}
      smartphoneSelected={smartphoneSelected}
    />
  );

describe("<FindSmartphoneModel />", () => {
  it("Should render without errors", () => {
    const { container } = renderComponent({ smartphoneSelected: null });
    expect(container).toMatchSnapshot();
  });

  describe("When the button 'Ver valores' is clicked", () => {
    it("Should trigger a track button event", () => {
      const { getByText } = renderComponent({ smartphoneSelected: null });

      fireEvent.click(getByText("Ver valores"));

      expect(smartphoneTracker.trackButton).toHaveBeenCalledWith(
        "Tabela Pipe Prices"
      );
    });

    it("Should invoke the toggleSmartphoneSelector fn", () => {
      const { getByText } = renderComponent({ smartphoneSelected: null });

      fireEvent.click(getByText("Ver valores"));

      expect(toggleSmartphoneSelector.mock.calls.length).toBe(1);
    });
  });

  describe("When the smartphoneSelected object is null", () => {
    it("Should render the component", () => {
      const { getByText } = renderComponent({ smartphoneSelected: null });

      getByText("Buscar modelo");
      getByText("Veja se sua proteção irá receber um reajuste");
      getByText("Ver valores");
    });
  });

  describe("When the smartphoneSelected object is not empty", () => {
    it("Should not render the component", () => {
      const { queryByText } = renderComponent({ smartphoneSelected: {} });

      expect(queryByText("Buscar modelo")).toBeFalsy();
      expect(
        queryByText("Veja se sua proteção irá receber um reajuste")
      ).toBeFalsy();
      expect(queryByText("Ver valores")).toBeFalsy();
    });
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });
});
