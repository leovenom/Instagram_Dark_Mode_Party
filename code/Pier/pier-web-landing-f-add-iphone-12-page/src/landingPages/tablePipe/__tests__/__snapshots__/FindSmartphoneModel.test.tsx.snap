// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FindSmartphoneModel /> Should render without errors 1`] = `
<div>
  <h2
    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 jPjwaE"
  >
    Buscar modelo
  </h2>
  <p
    class="Text__Component-sc-1g1y147-0 Text___StyledComponent-sc-1g1y147-1 jvkjLQ"
  >
    Veja se sua proteção irá receber um reajuste
  </p>
  <button
    class="Button__StyledButton-szjasf-0 eGmdcK"
  >
    Ver valores
  </button>
</div>
`;
